solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -1017,19 +1023,21 @@ public String getBaseURL() {
   }
 
   private static class AsyncTracker {
-    private static final int MAX_OUTSTANDING_REQUESTS = 1000;
+    private static final int DEFAULT_MAX_OUTSTANDING_REQUESTS = 1000;
```

Comment:
While we're at it, this would be a good use of `EnvUtils` like "solr.solrj.async.max.outstanding.requests"

Reply:
💯 -- it seems like this PR is motivated by a "custom solr client" use case? But unless we make this configurable in a regular deployment (as would be achieved by `EnvUtils`), there's no way to configure the clients that are used for intra-cluster communication, which was the concern that initially motivated the filing of SOLR-16932.

In fact, an `EnvUtils`-type approach was the _only_ approach taken in [the patch we've been using to mitigate this issue locally](https://github.com/cowpaths/fullstory-solr/pull/124).

Reply:
That sounds good, but I'd remove the `solrj` part of the sys prop

Reply:
Isn't the "solrj" part important though?  The component name following "solr" is important to categorize this setting.  Simply omitting "solrj" would make "async" the category, which is a bad name IMO.  I think "solrj" is a good category name and it should be chosen.  See https://cwiki.apache.org/confluence/display/SOLR/System+property+naming+structure
CC @janhoy